### PR TITLE
feat: tokio & handshake improvements

### PR DESCRIPTION
**Pull request by pavlix**

### CODE REVIEW

**Comment by mark-stopka** [comment]
I fixed the failing Windows build...

File: src/protocols/handshake.rs [hunk 1]
```diff
@@ -46,38 +48,117 @@ pub enum State {
...
+    }
+    pub fn build(&mut self) -> Result<HandshakeProtocol, Error> {
+        Ok(HandshakeProtocol {
+            role: self.role,
+            variant: self.variant,
+            network_magic: self.magic,
+            state: State::Propose,
+            result: None,
+        })
+    }
```
Comment (NicolasDP) [inline src/protocols/handshake.rs:99]: why add an error if no error can be returned ?

Comment (pavlix) [reply src/protocols/handshake.rs:99]: > why add an error if no error can be returned ?

That is in order to be future-proof. Builders can be extended without introducing a breaking change for consumer code. There might be validation needed in any of the builders in the future.

Comment (mark-stopka) [reply src/protocols/handshake.rs:99]: @NicolasDP do you consider this satisfactory answer, if so I am ready to merge it in...

**Review by mark-stopka — APPROVED** [review]

**Review by mark-stopka — COMMENTED** [review]
LGTM

**Pull request merged**